services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py

```diff
@@ -287,21 +273,33 @@ async def download_target_ports(
     )
 
     # let's gather all the data
-    download_tasks: deque[Coroutine[Any, int, Any]] = deque()
-    for port_value in (await getattr(PORTS, port_type_name.value)).values():
-        # if port_keys contains some keys only download them
-        logger.debug("Checking node %s", port_value.key)
-        if port_keys and port_value.key not in port_keys:
-            continue
-        # collect coroutines
-        download_tasks.append(_get_data_from_port(port_value))
-    logger.debug("retrieving %s data", len(download_tasks))
-
-    data, transferred_bytes = await _download_files(target_path, download_tasks)
+    ports_to_get = [
+        port_value
+        for port_value in (await getattr(PORTS, port_type_name.value)).values()
+        if (not port_keys) or (port_value.key in port_keys)
+    ]
+    async with progress_bar.sub_progress(steps=len(ports_to_get)) as sub_progress:
+        results = await logged_gather(
+            *[
+                _get_data_from_port(
+                    port, target_dir=target_dir, progress_bar=sub_progress
+                )
+                for port in ports_to_get
+            ],
+            max_concurrency=4,
```

Comment:
Maybe we want to limit this to 2. If we have lots of sidecars on the same node pulling at the same time. I think we already have this policy in other places.

Reply:
ok for 2.